private/model/api/api.go

```diff
@@ -933,6 +940,13 @@ const (
 			"{{ $s.ErrorName }}": newError{{ $s.ShapeName }},
 		{{- end }}
 	}
+	{{- if .Metadata.AWSQueryCompatible }}
+	var queryExceptionFromCode = map[string]func(protocol.ResponseMetadata, string)error {
```

Comment:
why do we need separate logic here? it looks like my biggest confusion in all my comments is why arent we just doing an override on the error code right at the point where we unmarshal the raw http response? and then use all the existing error typed logic?

Reply:
This logic here creates a map of error code to error function with error code compatibility